src/core/lib/gprpp/memory.h

```diff
@@ -46,6 +46,17 @@ class DefaultDeleteChar {
 template <typename T>
 using UniquePtr = std::unique_ptr<T, DefaultDeleteChar>;
 
+template <class T>
```

Comment:
Can we omit this variant? It seems easy to misuse.

Reply:
Yeah, it's meant to support something like `grpc_channel_stack` which the tail can have various size but it's not majority so it can be omitted. If needed later, it will be added with a different name to avoid a mistake.